Update for embedded-hal 1.0

madmo opened the pull request:
Updated all dependencies to work with embedded-hal version 1.0

madmo committed:
Update for embedded-hal 1.0

eldruin reviewed (commented):
Nice, thanks!
Could you also:
- Add an entry to the changelog about this?
- Fix the formatting?
- Update the MSRV to Rust 1.62.0?

madmo committed:
Update workflow with new MSRV version

madmo committed:
Update CHANGELOG.md

madmo commented:
> Nice, thanks! Could you also:
> 
> * Add an entry to the changelog about this?
> * Fix the formatting?
> * Update the MSRV to Rust 1.62.0?

Sure, pushed the new changes. Thanks for the feedback :+1:

eldruin commented:
Thanks! So that we get the CI to pass, could you add an underscore in these [unused arguments in the test](https://github.com/eldruin/opt300x-rs/actions/runs/8018324731/job/21905377062?pr=1) and also update the CI clippy version to 1.62.0?

madmo committed:
Fix clippy warnings

madmo committed:
Use cargo-llvm-cov for coverage

cargo-llvm-cov is more acurate than tarpaulin

madmo commented:
Ok, now I fixed all warnings, all green: https://github.com/madmo/opt300x-rs/actions/runs/8021089068

Replaced cargo-tarpulin with cargo-llvm-cov, because the tarpulin action seems to be broken/abdoned.

eldruin commented on Cargo.toml:
```diff
@@ -21,12 +21,12 @@ include = [
-linux-embedded-hal = "0.3"
-embedded-hal-mock = "0.8"
+linux-embedded-hal = "0.4.0"
+embedded-hal-mock = "0.10.0"
```
```suggestion
embedded-hal-mock = {version="0.10.0", default-features=false, features=["eh1"]}
```
So that we avoid the dependency on `embedded-hal` 0.2

eldruin reviewed (commented):
Great, looks good to me!
Do you have any reason to set the MSRV in CI to 1.63.0 instead of 1.62.0 that I should be aware of? or was it only a typo?

eldruin committed:
Avoid embedded-hal 0.2 dependency

madmo commented:
> Great, looks good to me! Do you have any reason to set the MSRV in CI to 1.63.0 instead of 1.62.0 that I should be aware of? or was it only a typo?

The tests were not compiling for me under 1.62.0 because of embedded-hal-mock v0.10.0 which requires 1.63.0.
I thought 1.62.0 was a typo.

eldruin commented on .github/workflows/build.yml:
```diff
@@ -13,7 +13,7 @@ jobs:
     strategy:
       matrix:
-        rust: [stable, 1.31.0]
+        rust: [stable, 1.63.0]
```
```suggestion
        rust: [stable, 1.62.0]
```

eldruin reviewed (commented):
Ah, I see. Normally in the CI I run the tests only on stable due to these kind of issues with dev-dependencies and only build this crate with the selected MSRV.

eldruin committed:
Use 1.62.0 MSRV

eldruin reviewed (approved):
Great, thank you for your work!

eldruin commented:
For the record, I published this in [version 1.0.0](https://github.com/eldruin/opt300x-rs/releases/tag/v1.0.0)
